File: Examples/CSharp/ModifyingAndConvertingImages/MetaFiles/ConvertEMFToPDF.cs
using System;
using System.IO;
using System.Drawing;
using Aspose.Imaging.CoreExceptions;
using Aspose.Imaging.FileFormats.Emf;
using Aspose.Imaging.ImageOptions;

/*
This project uses the Automatic Package Restore feature of NuGet to resolve the Aspose.Imaging for .NET API reference 
when the project is built. Please check https://learn.microsoft.com/en-us/nuget/resources/nuget-faq for more information. 
If you do not wish to use NuGet, you can manually download Aspose.Imaging for .NET API from https://releases.aspose.com/, 
install it, and then add its reference to this project. For any issues, questions, or suggestions, 
please feel free to contact us using https://forum.aspose.com/
*/

namespace Aspose.Imaging.Examples.CSharp.ModifyingAndConvertingImages.MetaFiles
{
    class ConvertEMFToPDF
    {
        public static void Run()
        {
            //ExStart:ConvertEMFToPDF
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_MetaFiles();

            string[] filePaths = {
                "Picture1.emf"
            };

            Console.WriteLine("Running example ConvertEMFToPDF");

            foreach (string filePath in filePaths)
            {
                string outPath = dataDir + filePath + "_out.pdf";
                using (var image = (EmfImage)Image.Load(dataDir + filePath))
                using (FileStream outputStream = new FileStream(outPath, FileMode.Create))
                {
                    if (!image.Header.EmfHeader.Valid)
                    {
                        throw new ImageLoadException(string.Format("The file {0} is not valid", outPath));
                    }

                    EmfRasterizationOptions emfRasterization = new EmfRasterizationOptions
                    {
                        PageWidth = image.Width,
                        PageHeight = image.Height,
                        BackgroundColor = Color.WhiteSmoke
                    };
                    PdfOptions pdfOptions = new PdfOptions
                    {
                        VectorRasterizationOptions = emfRasterization
                    };
                    image.Save(outputStream, pdfOptions);
                }
            }

            Console.WriteLine("Finished example ConvertEMFToPDF");
            //ExEnd:ConvertEMFToPDF
        }
    }
}

File: Examples/VisualBasic/ModifyingAndConvertingImages/MetaFiles/ConvertEMFToPDF.vb
Imports System.IO
Imports Aspose.Imaging.CoreExceptions
Imports Aspose.Imaging.FileFormats.Emf
Imports Aspose.Imaging.ImageOptions

'
'This project uses Automatic Package Restore feature of NuGet to resolve Aspose.Imaging for .NET API reference 
'when the project is build. Please check https://Docs.nuget.org/consume/nuget-faq for more information. 
'If you do not wish to use NuGet, you can manually download Aspose.Imaging for .NET API from http://www.aspose.com/downloads, 
'Install it and then add its reference to this project. For any issues, questions or suggestions 
'Please feel free to contact us using http://www.aspose.com/community/forums/default.aspx
'

Namespace Aspose.Imaging.Examples.VisualBasic.ModifyingAndConvertingImages.MetaFiles
    Class ConvertEMFToPDF
        Public Shared Sub Run()
            ' ExStart:ConvertEMFToPDF
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_MetaFiles()

            Dim filePaths As String() = {"Picture1.emf"}

            For Each filePath As String In filePaths
                Dim outPath As String = (dataDir & filePath) + "_out.pdf"

                Using image__1 = DirectCast(Image.Load(dataDir & filePath), EmfImage)
                    Using outputStream As New FileStream(outPath, FileMode.Create)
                        If Not image__1.Header.EmfHeader.Valid Then
                            Throw New ImageLoadException(String.Format("The file {0} is not valid", outPath))
                        End If

                        Dim emfRasterization As New EmfRasterizationOptions()
                        emfRasterization.PageWidth = image__1.Width
                        emfRasterization.PageHeight = image__1.Height
                        emfRasterization.BackgroundColor = Color.WhiteSmoke

                        Dim pdfOptions As New PdfOptions()
                        pdfOptions.VectorRasterizationOptions = emfRasterization

                        image__1.Save(outputStream, pdfOptions)
                    End Using
                End Using
            Next
            ' ExEnd:ConvertEMFToPDF
        End Sub
    End Class
End Namespace